packages/mask/src/plugins/Tips/SNSAdaptor/bodyViews/AddWallet.tsx

```diff
@@ -0,0 +1,170 @@
+import { BindingProof, NextIDAction, NextIDPlatform } from '@masknet/shared-base'
+import { WalletMessages } from '@masknet/plugin-wallet'
+import { useRemoteControlledDialog } from '@masknet/shared-base-ui'
+import {
+    isSameAddress,
+    ProviderType,
+    useAccount,
+    useChainId,
+    useNetworkType,
+    useProviderType,
+    useWallet,
+} from '@masknet/web3-shared-evm'
+import { memo, useCallback, useEffect, useState } from 'react'
+import { SignSteps, Steps } from '../../../../components/shared/VerifyWallet/Steps'
+import { useAsync, useAsyncFn } from 'react-use'
+import { NextIDProof } from '@masknet/web3-providers'
+import Services from '../../../../extension/service'
+import { useCustomSnackbar } from '@masknet/theme'
+import formatDateTime from 'date-fns/format'
+import { useProviderDescriptor, useReverseAddress, useWeb3State } from '@masknet/plugin-infra/web3'
+import { useI18N } from '../../../../utils'
+import { PluginId } from '@masknet/plugin-infra'
+
+interface AddWalletViewProps {
+    currentPersona: any
+    bounds: BindingProof[]
+    onCancel: () => void
+}
+
+const AddWalletView = memo(({ currentPersona, bounds, onCancel }: AddWalletViewProps) => {
+    const { t } = useI18N()
+    const [isBound, setIsBound] = useState(false)
+    const [signed, setSigned] = useState(false)
+    const { showSnackbar } = useCustomSnackbar()
+    const curProviderType = useProviderType()
```

Comment:
There is no `providerType` variable here.

```suggestion
    const providerType = useProviderType()
```